students/zach_meves/lesson03/basic_operations.py

```diff
@@ -0,0 +1,96 @@
+"""
+Basic database operations.
+"""
+
+from customer_model import Customer, DB
+
+
+def add_customer(customer_id: int, name: str, lastname: str, home_address: str,
+                 phone_number: str, email_address: str, status: bool, credit_limit: float):
+    """
+    Add a new customer to the database.
+
+    :param customer_id: int, Customer ID
+    :param name: str, First name
+    :param lastname: str, Last name
+    :param home_address: str, Home address
+    :param phone_number: str, Phone number
+    :param email_address: str, Email address
+    :param status: bool, True for "active", False for "inactive"
+    :param credit_limit: float, credit limit
+    """
+
+    # Create a new customer object
+    with DB.transaction():
+        try:
+            cust = Customer.create(id=customer_id, name=name, last_name=lastname,
+                                   address=home_address, phone=phone_number, email=email_address,
+                                   status=status, credit_limit=credit_limit)
+        except Exception as err:
+            raise ValueError(err)
+        else:
+            cust.save()
+
+
+def search_customer(customer_id: int) -> dict:
+    """
+    Return a dictionary representing the customer with the given ID.
+    :param customer_id: int, customer ID
+    :return: dict
+    """
+
+    try:
+        res = _get_by_id(customer_id)
+    except Exception:
+        return {}
+    else:
+        return {'name': res.name, 'last_name': res.last_name, 'email': res.email,
+                'phone': res.phone}
+
+
+def delete_customer(customer_id: int):
+    """
+    Delete customer from database.
+
+    :param customer_id: int, customer ID
+    """
+
+    try:
+        res = _get_by_id(customer_id)
+    except Exception:
+        pass
+    else:
+        res.delete_instance()
+
+
+def update_customer_credit(customer_id: int, credit_limit: float):
+    """
+    Updates an existing customer's credit limit. Raises a ValueError if no such
+    customer exists.
+
+    :param customer_id: int, customer ID
+    :param credit_limit: float, new credit limit
+    """
+
+    try:
+        res = _get_by_id(customer_id)
+    except Exception:
+        raise ValueError(f"No customer with ID {customer_id}")
+    else:
+        with DB.transaction():
+            res.credit_limit = credit_limit
+            res.save()
+
+
+def list_active_customers() -> int:
+    """
+    Return number of customers whose status is currently active.
+
+    :return: int
+    """
+
+    return Customer.select().where(Customer.status == True).count()
```

Comment:
Because of the concept of _Truthiness_ in Python, you don't need to have _== True_, it suffices to do:

`return Customer.select().where(Customer.status).count()`